lntest/wait/wait.go

```diff
@@ -58,6 +58,13 @@ func NoError(f func() error, timeout time.Duration) error {
 	// If f() doesn't succeed within the timeout, return the last
 	// encountered error.
 	if err := Predicate(pred, timeout); err != nil {
+		// Handle the case where the passed in method, f, hangs for the
+		// full timeout
+		if predErr == nil {
```

Comment:
could `f` succeed but `Predicate` still return an error? Seems that currently we don't care about what `err` (returned by `Predicate`) is

Reply:
If `f` is run (and completes) before `Predicate` returns an error then `Predicate` will return nil & `predErr` will be set to whatever `f` returned. Thus, if `f` succeeds, then no, `Predicate` will not return an error